src/auth/session.service.ts

```diff
@@ -153,6 +185,21 @@ export class SessionService {
       'auth/session:revoke',
       sessionId,
     );
-    await this.sessionRepository.update({ id: sessionId }, { revoked: true });
+    const ret = await this.sessionRepository.update(
+      { id: sessionId },
+      { revoked: true },
+    );
+    /* istanbul ignore if */
+    if (ret.affected !== 1) {
+      throw new Error(
+        `In an attempt to revoke session with id ${sessionId},\n` +
+          `the refresh token is valid, but the session does not exist.\n` +
+          `Here are three possible reasons:\n` +
+          `1. There is a bug in the code.\n` +
+          `2. The database is corrupted.\n` +
+          `3. We are under attack.\n` +
```

Comment:
No. Service should never be stopped because of an error.

This is what is called availability. Service should always be available, even if there is an error.

Data protection should be done with database authorization management.

Reply:
Availability only make sense when there is no serious circumstances like under attack or database corruption.
While high availability is a crucial goal, it's equally important to recognize scenarios where maintaining availability might not be the best course of action. Balancing availability with other factors like security and data integrity is key to ensuring that the service remains robust and reliable.
We could continue to discuss this in more depth in the future.